ethcore/src/engines/authority_round/finality.rs

```diff
@@ -52,28 +52,31 @@ impl RollingFinality {
 	///
 	/// Fails if any provided signature isn't part of the signers set.
 	pub fn build_ancestry_subchain<I>(&mut self, iterable: I) -> Result<(), UnknownValidator>
-		where I: IntoIterator<Item=(H256, Address)>
+		where I: IntoIterator<Item=(H256, Vec<Address>)>
 	{
 		self.clear();
-		for (hash, signer) in iterable {
-			if !self.signers.contains(&signer) { return Err(UnknownValidator) }
+		for (hash, signers) in iterable {
+			if !signers.iter().all(|s| self.signers.contains(s)) { return Err(UnknownValidator) }
```

Comment:
using `any` and putting the negation inside the closure will short-circuit as soon as the first unknown is detected